### PR TITLE
osd/ReplicatedBackend: reuse the already computed cost

### PR DESCRIPTION
**Pull request by runsisi**
no need to compute the cost a second time

Signed-off-by: runsisi <runsisi@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
looks good

File: src/messages/MOSDPGPush.h [hunk 1]
```diff
@@ -43,6 +43,10 @@ class MOSDPGPush : public Message {
     return cost;
   }
 
+  void set_cost(uint64_t c) {
```
Comment (tchaikov) [inline src/messages/MOSDPGPush.h]: cost is a public member variable. if we aways access it via its accessors, maybe we can mark it private?

Comment (xiexingguo) [inline src/messages/MOSDPGPush.h]: @tchaikov  Good idea. Withdraw the needs-qa label for now:-)

**Comment by runsisi** [comment]
@tchaikov @xiexingguo done, thank you both~

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2016-05-13_16:00:42-rados-wip-yuri-testing-distro-basic-smithi/

**Pull request merged**